docs/testing.md

```diff
@@ -39,20 +39,54 @@ They generally follow TPCH but are not guaranteed to be the same.
 
 ## Unit tests
 
-Unit tests exist in the tests directory. This is unconventional and is done so we can run the tests
-on the final shaded version of the plugin. It also helps with how we collect code coverage.
+Unit tests exist in the [here](tests) directory. This is unconventional and is done so we can run
+ the tests on the final shaded version of the plugin. It also helps with how we collect code coverage.
+
+in order to run the unit-tests follow these steps
+1. issue the maven command to run the tests with `mvn test`. this will run all the tests
+2. to run individual tests append `-dwildcardsuites=<comma separated list of wildcard suite names to execute>` to the above command 
+
+for more information about using scalatest with maven please refere [here](https://www.scalatest.org/user_guide/using_the_scalatest_maven_plugin)
+    
+#### Running unit-tests against specific apache spark versions. 
 You can run the unit tests against different versions of Spark using the different profiles. The
-default version runs again Spark 3.0.0, to run against other version use one of the following profiles:
-   - `-Pspark301tests` (Spark 3.0.1)
-   - `-Pspark302tests` (Spark 3.0.2)
-   - `-Pspark310tests` (Spark 3.1.0)
+default version runs against Spark 3.0.0, to run against other versions use one of the following
+ profiles:
+   - `-Pspark301tests` (spark 3.0.1)
+   - `-Pspark302tests` (spark 3.0.2)
+   - `-Pspark310tests` (spark 3.1.0)
+
+Please refer to the [pom.xml](tests/pom.xml) to see the list of profiles supported
+apache spark specific configurations can be passed in by setting environment-variable spark_conf 
+
+Examples: 
+-To run tests against Apache Spark 3.1.0, 
+ `mvn -P spark310tests test -DwildcardSuites="com.nvidia.spark.rapids.ParquetWriterSuite"`.
+- To pass Apache Spark configs `--conf spark.dynamicAllocation.enabled=false --conf spark.task.cpus=1` do something like.
+ `SPARK_CONF="spark.dynamicAllocation.enabled=false,spark.task.cpus=1" mvn ...`
+- To run test ParquetWriterSuite in package com.nvidia.spark.rapids, issue `mvn test -DwildcardSuites="com.nvidia.spark.rapids.ParquetWriterSuite"`
 
 ## Integration tests
 
-Integration tests are stored in the [integration_tests](../integration_tests/README.md) directory.
-There are two frameworks used for testing. One is based off of pytest and pyspark in the 
-`src/main/python` directory. These tests will run as a part of the maven build if you have the environment
-variable `SPARK_HOME` set.
+Integration tests are stored in the [integration_tests](integration_tests) directory. 
```

Comment:
This is not in the right directory for this link to work.  Please change it back to the README or at least ../integration_tests